presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -110,10 +110,16 @@ private boolean exploreNode(int group, Context context)
             done = true;
             for (Rule rule : rules) {
                 long start = System.nanoTime();
-                Optional<PlanNode> transformed = rule.apply(node, context.getLookup(), context.getIdAllocator(), context.getSymbolAllocator());
-                long duration = System.nanoTime() - start;
-
-                stats.record(rule, duration, transformed.isPresent());
+                Optional<PlanNode> transformed = null;
+                try {
+                    transformed = rule.apply(node, context.getLookup(), context.getIdAllocator(), context.getSymbolAllocator());
+                    long duration = System.nanoTime() - start;
```

Comment:
Ok, I wasn't crazy... I actually had done it that way, but forgot to push --force over my branch before I submitted the PR.